wpilib/src/gyro.rs

```diff
@@ -0,0 +1,134 @@
+use wpilib_sys::*;
+
+pub trait Gyro {
+    /// Calibrate the gyro by running for a number of samples and computing the
+    /// center value. Then use the center value as the Accumulator center value for
+    /// subsequent measurements. It's important to make sure that the robot is not
+    /// moving while the centering calculations are in progress, this is typically
+    /// done when the robot is first turned on while it's sitting at rest before
+    /// the competition starts.
+    fn calibrate(&self) -> HalResult<()>;
```

Comment:
I think this should have an associated type `Error` instead, so the methods here would return `Result<(), Self::Error>`.